### PR TITLE
Converts slapcrafting into a component

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Converts slapcrafting into a component!

The component is added on to an ingredient (presumably the main ingredient) with a list of recipes attached. If you interact an ingredient (if no ignredients, a tool) with it, you will start crafting the recipe. If there's multiple, pick between them with a radial menu.

Opening on draft as there's just a liiiiil bit left to do. The actual wired rod was left for last.

## Why It's Good For The Game

Slapcrafting is simply better and more accessible and less laggy than menu crafting. By making it a component we can attach it to things in which it'd make sense to while stopping unintended weirdness that might arise from this being global.

Additionally the way examine lets you see crafting recipes opens up visibility for those, which allows new players to learn about them in a intuitive manner.

## Changelog

:cl:
refactor: Turned slapcrafting into a component! You can examine compatible items to see what recipes they can be used in, and what the ingredients for them are. For example, spears and the head-on-spear crafting recipe.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
There are items that act like crafting but aren't - the current obvious example is wired rods, spears, stunprods. I would greatly appreciate if these were pointed out so I could convert them to crafting recipes.

**Comment by Iamgoofball** [comment]
@MrStonedOne does this pass muster?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by Colovorat** [comment]
>to see what recipes they can be used in

i'm actually curious on how many there recipes i don't know, like shiv isn't very obvious

**Comment by carlarctg** [comment]
Oh shiv is one of those fake crafting recipes, I think. Thank you, I'll update that

File: code/datums/components/crafting/guncrafting.dm [hunk 1]
```diff
@@ -2,12 +2,24 @@
...
 /obj/item/weaponcrafting/receiver
 	name = "modular receiver"
 	desc = "A prototype modular receiver and trigger assembly for a firearm."
 	icon = 'icons/obj/weapons/improvised.dmi'
 	icon_state = "receiver"
 
+/obj/item/weaponcrafting/receiver/create_slapcraft_component()
+	AddComponent(/datum/component/slapcrafting,\
+			slapcraft_recipes = list(/datum/crafting_recipe/pipegun)\
+	)
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/datums/components/crafting/guncrafting.dm:21]: Handled with static vars per discord convo

**Comment by Jacquerel** [comment]
well nobody else seems to have opinions about this so lets get it merged i suppose

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -302,69 +302,6 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 /obj/item/katana/cursed //used by wizard events, see the tendril_loot.dm file for the miner one
 	slot_flags = null
 
-/obj/item/wirerod
```
Comment (Jacquerel) [inline code/game/objects/items/weaponry.dm:305]: oh I should have asked you to make an updatepaths script for this :clueless:

**Pull request merged**